### PR TITLE
#110 updating ingress config to fix websocket connectivity.

### PR DESCRIPTION
**Pull request by DovOps**
Added websocket support where needed to nginx 

THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE FINOS Corporate Contributor License Agreement.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: ingress/nginx.traderx.conf.template [hunk 1]
```diff
@@ -1,32 +1,52 @@
...
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade ${DOLLAR}http_upgrade;
+            proxy_set_header Connection "upgrade";
+        }
+
+        location /trade-feed/index.html {
+            proxy_pass http://trade-feed:18086/;
+        }
+
+        location /trade-feed/ {
```
Comment (maoo) [inline ingress/nginx.traderx.conf.template:23]: we should add an "OR /socket.io"

File: ingress/nginx.traderx.conf.template [hunk 2]
```diff
@@ -1,32 +1,52 @@
...
+            proxy_set_header X-Forwarded-For ${DOLLAR}proxy_add_x_forwarded_for;
+            proxy_set_header Host ${DOLLAR}http_host;
+
+            proxy_pass http://trade-feed:18086/trade-feed/;
```
Comment (maoo) [inline ingress/nginx.traderx.conf.template:27]: remove sub-path

File: ingress/nginx.traderx.conf.template [hunk 3]
```diff
@@ -1,32 +1,52 @@
...
+        }
+
+        location /ng-cli-ws {
+            proxy_pass http://web-front-end-angular:18093/ng-cli-ws;
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade ${DOLLAR}http_upgrade;
+            proxy_set_header Connection "upgrade";
+        }
+
+        location /trade-feed/index.html {
```
Comment (maoo) [inline ingress/nginx.traderx.conf.template:19]: this is not needed

File: ingress/nginx.traderx.conf.template [hunk 4]
```diff
@@ -1,32 +1,52 @@
...
-}
\ No newline at end of file
+        location /db-web/ {
+            proxy_pass http://database:18084/;
+        }
+        location /reference-data/ {
+            proxy_pass http://reference-data:18085/;
+        }
+
+        location /ng-cli-ws {
```
Comment (maoo) [inline ingress/nginx.traderx.conf.template:12]: this is needed

**Comment by DovOps** [comment]
@maoo  - Try this branch now - The entire app works, as well as /trade-feed/ (for the standalone message viewer), without requiring a non-default path for the feed.

**Comment by maoo** [comment]
> @maoo - Try this branch now - The entire app works, as well as /trade-feed/ (for the standalone message viewer), without requiring a non-default path for the feed.

Redeployed, but I still see errors; FYI - the nginx.conf on EC2 only contains the following logic:
```
        location / {
                proxy_pass http://localhost:8080/;
        }
```

Do we need to make any change here as well?

**Pull request merged**